packages/ui/src/views/connectors/connector-entity-form.tsx

```diff
@@ -162,14 +162,16 @@ export const ConnectorEntityForm: FC<ConnectorEntityFormProps> = ({
                 </Alert.Root>
               )}
             </EntityFormLayout.Form>
-          </Inner>
+          </Body>
           <Footer>
-            {isCreate && (
-              <Button variant="outline" onClick={() => onBack?.()}>
-                Back
-              </Button>
-            )}
-            <Button onClick={() => rootForm.submitForm()}>{isCreate ? 'Submit' : 'Apply changes'}</Button>
+            <ButtonGroup>
+              {isCreate && !!onBack && (
+                <Button variant="outline" onClick={() => onBack?.()}>
```

Comment:
At this point you've proven `onBack` exists, so you don't need the optional chaining. You can probably call it directly instead of wrapping it in an anonymous function.